### DISCUSSION TITLE
Creating Filled Stadiums

### DISCUSSION THREAD

**Discussion by johnthagen** [mozman/ezdxf#1193] [Q&A] [answered]
Is it possible to natively define a filled stadium object within DXF?

- https://en.wikipedia.org/wiki/Stadium_(geometry)

<img src="https://github.com/user-attachments/assets/076639bb-9781-47ee-bd7b-8ef958ed4923" alt="image" width="300" height="570">

Note this is similar to an ellipse:
- https://github.com/mozman/ezdxf/discussions/1161#discussioncomment-10621558

But the geometric shape is different.

**Comment by mozman** [comment, marked as answer]
A HATCH entity to get the filling, a LWPOLYLINE entity to get the border line if required.

```Python
import ezdxf
from ezdxf.layouts import Modelspace

def add_stadium_as_lwpolyline(msp: Modelspace, a: float, r: float) -> None:
    # format: x, y, bulge
    # bulge value always at the start vertex of the arc
    d = 2 * r
    points = [
        (0, 0, 0),
        (0, a, 1),  # semi-circle follows
        (-d, a, 0),
        (-d, 0, 1),  # semi-circle follows
        # close shape
    ]
    msp.add_lwpolyline(points, format="xyb", close=True)


def add_stadium_as_hatch(msp: Modelspace, a: float, r: float) -> None:
    d = 2 * r
    hatch = msp.add_hatch(color=1)
    edges = hatch.paths.add_edge_path()
    edges.add_line((0, 0), (0, a))
    edges.add_arc(center=(-r, a), radius=r, start_angle=0, end_angle=180)
    edges.add_line((-d, a), (-d, 0))
    edges.add_arc(center=(-r, 0), radius=r, start_angle=180, end_angle=0)


def main():
    doc = ezdxf.new()
    msp = doc.modelspace()
    add_stadium_as_hatch(msp, 10, 5)
    add_stadium_as_lwpolyline(msp, 10, 5)
    doc.saveas("stadium.dxf")


if __name__ == "__main__":
    main()
```

![image](https://github.com/user-attachments/assets/f1fb6eaa-caf0-4092-b0e4-e428281e2825)

**Reply by johnthagen** [reply]
This worked great. One thing to note for others, it's important that, like this example, you draw each line and arc in a counterclockwise order around the perimeter.

**Reply by mozman** [reply]
It is possible to use clockwise arcs. The `add_arc()` method has an argument called `ccw` and is `True`  by default, set it `False` to draw a clockwise arc. This feature is only supported for arcs in boundary paths the regular `ARC` entity doesn't support (and doesn't need) such a flag.
